src/pointer-analysis/goto_program_dereference.h

```diff
@@ -36,7 +36,7 @@ class goto_program_dereferencet:protected dereference_callbackt
     : options(_options),
       ns(_ns),
       value_sets(_value_sets),
-      dereference(_ns, _new_symbol_table, *this, ID_nil, false)
+      dereference(_ns, _new_symbol_table, *this, ID_nil, false, messaget())
```

Comment:
❗ Passing a default `messaget` instance doesn't look right here. The instance should come from the caller.

Reply:
Is this a valid construct? The points-to sets are displayed during symex only. `exprt dereference(const exprt &pointer, bool display_points_to_sets = false);` sets `display_points_to_set` to false by default so any calls to `value_set_dereferencet::dereference()` from the goto program conversion phase does not display the points-to sets. Similarly, is it okay to set the `messaget` here to a default message instance for `goto_program_dereferencet`?
```
goto_program_dereferencet(
    const namespacet &_ns,
    symbol_tablet &_new_symbol_table,
    const optionst &_options,
    value_setst &_value_sets,
    const messaget &_log = messaget())
    : options(_options),
      ns(_ns),
      value_sets(_value_sets),
      dereference(_ns, _new_symbol_table, *this, ID_nil, false, _log)
  {
  }
```

Passing a valid messaget object here`goto_program_dereferencet` will need a lot of additional refactoring across the goto program conversion code modules.

Reply:
Ok, thanks for the explanation. Please put a code comment to clarify why a default messaget instance is passed in this place. The next person stumbling over that code should not have to redo your reasoning. Many thanks!

Reply:
Should I retain the current implementation i.e `dereference(_ns, _new_symbol_table, *this, ID_nil, false, messaget())` or replace it with the construct above? 
I will add in the comment.

Reply:
The one above is probably slightly cleaner as it pushes the messaget further up. But, still, please put a comment in goto_program_dereferencet then.

Reply:
Fixed. Comment added.